velox/functions/prestosql/aggregates/AverageAggregate.cpp

```diff
@@ -223,50 +236,72 @@ class AverageAggregate : public exec::Aggregate {
         baseRowVector->childAt(0)->template as<SimpleVector<TAccumulator>>();
     auto baseCountVector =
         baseRowVector->childAt(1)->template as<SimpleVector<int64_t>>();
-
-    if (decodedPartial_.isConstantMapping()) {
-      if (!decodedPartial_.isNullAt(0)) {
-        auto decodedIndex = decodedPartial_.index(0);
-        if constexpr (checkNullFields) {
-          VELOX_USER_CHECK(
-              !baseSumVector->isNullAt(decodedIndex) &&
-              !baseCountVector->isNullAt(decodedIndex));
-        }
-        auto count = baseCountVector->valueAt(decodedIndex);
-        auto sum = baseSumVector->valueAt(decodedIndex);
+    if (decodedPartial_.isIdentityMapping() && baseSumVector &&
```

Comment:
do we have a benchmark for this agg? I wonder if we can just structure the cases like 
case1 : (decodedPartial_.isConstantMapping())
case2: (decodedPartial_.isIdentityMapping())
case3 : ELSE

Then create decodedVector for the children in case 2 and 3.
The reason I ask about benchmark is because creating a decodedVector for flat should be fairly lightweight as i directly points to the values and nulls buffers without doing any computation. So we can skip the special case for when both children are flat and ensure using the benchmark that we do not regress in perf.

Reply:
According to offline discussion, I updated the code to decode child vectors in all cases.